float32 length
float32 duration
int16   turn